From 65d0109830113c4e79896e01b90904de9b8aad98 Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Tue, 16 Oct 2018 03:28:34 -0700
Subject: [PATCH 0598/1200] octeontx: mtd: Add mtd partitioning support

This patch enables MTD partitioning support.  A default partition table
has also been added which defines 6MB for the boot partition, 9MB for
data, and 64K for the environment, since 64K is the minimum supported
size.  I do not know what is after the environment.

Note that it is possible for U-Boot to add the partitioning to the
device tree so it can be passed on to Linux but this is not enabled
at this time.

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 configs/octeontx_81xx_defconfig | 5 +++++
 include/configs/octeontx_81xx.h | 9 ++++-----
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/configs/octeontx_81xx_defconfig b/configs/octeontx_81xx_defconfig
index 5f078ad647..2b97f2fb19 100644
--- a/configs/octeontx_81xx_defconfig
+++ b/configs/octeontx_81xx_defconfig
@@ -62,7 +62,12 @@ CONFIG_MISC=y
 CONFIG_DM_MMC=y
 CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
+CONFIG_MTD_PARTITIONS=y
+CONFIG_CMD_MTDPARTS=y
+CONFIG_MTDPARTS_DEFAULT="mtdparts=octeontx_nor0:6m@0(boot)ro,9m@0x600000(data),64k@0xf00000(env)ro\0"
+CONFIG_MTDIDS_DEFAULT="nor0=octeontx_nor0"
 CONFIG_DM_SPI_FLASH=y
+CONFIG_SPI_FLASH_MTD=y
 CONFIG_SPI_FLASH=y
 CONFIG_SPI_FLASH_4BAIS=y
 CONFIG_SPI_FLASH_MACRONIX=y
diff --git a/include/configs/octeontx_81xx.h b/include/configs/octeontx_81xx.h
index 7bd15a1cea..f25e5c34d4 100644
--- a/include/configs/octeontx_81xx.h
+++ b/include/configs/octeontx_81xx.h
@@ -78,10 +78,10 @@
 
 /***** SPI Defines *********/
 #ifdef CONFIG_DM_SPI_FLASH
-#define CONFIG_SF_DEFAULT_SPEED 12500000
-#define CONFIG_SF_DEFAULT_MODE	0
-#define CONFIG_SF_DEFAULT_BUS	0
-#define CONFIG_SF_DEFAULT_CS	0
+# define CONFIG_SF_DEFAULT_SPEED 12500000
+# define CONFIG_SF_DEFAULT_MODE	0
+# define CONFIG_SF_DEFAULT_BUS	0
+# define CONFIG_SF_DEFAULT_CS	0
 #endif
 
 /** Extra environment settings */
@@ -125,7 +125,6 @@
 #define CONFIG_NAND_OCTEONTX
 #define CONFIG_MTD_DEVICE
 /*#define CONFIG_MTD_CONCAT*/
-#define CONFIG_MTD_PARTITIONING
 #define CONFIG_SYS_MAX_NAND_DEVICE	8
 #define CONFIG_SYS_NAND_SELF_INIT
 #define CONFIG_SYS_NAND_ONFI_DETECTION
-- 
2.22.0

